File: typescript/src/shapes/circle.ts
function circle(radius: number): Shape {
    return {
        getWidth: function (): number {
            return 2 * radius
        },
        getHeight: function (): number {
            return 2 * radius
        },
        computeArea: function (): number {
            return Math.PI * radius * radius
        }
    }
}

export { circle }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Circle.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Circle implements Shape {
    private final double radius;

    public Circle(double radius) {
        this.radius = radius;
    }

    public double getArea() {
        return Math.PI * getWidth() * getHeight() / 4.0;
    }

    public double getWidth() {
        return 2.0 * this.radius;
    }

    public double getHeight() {
        return 2.0 * this.radius;
    }
}
